mne/beamformer/_dics.py

```diff
@@ -239,6 +244,14 @@ def make_dics(info, forward, csd, reg=0.05, label=None, pick_ori=None,
                         (freq, i + 1, n_freqs))
 
         Cm = csd.get_data(index=i)
+        if noise_csd is not None:
+            noise_csd_freq = noise_csd.get_data(index=i, as_cov=True)
+        else:
+            noise_csd_freq = None
+
+        _, _, proj, vertices, G, _, nn, orient_std = _prepare_beamformer_input(
+            info, forward, label, pick_ori, noise_cov=noise_csd_freq,
+            combine_xyz=combine_xyz, exp=exp)
```

Comment:
Does this really need to be done per frequency now? It was outside the loop before if I see it right.

Reply:
Oh wait ... we now whiten each lead field differently, is that it? Hummmmm, is that something we want (I do understand why we _have_ it)? Different lead fields per frequency ... ? It strikes me as an odd thing to have. One step further: do we actually have to whiten the lead field or would it be sufficient to whiten the data covariance /CSD matrix.

Reply:
We absolutely have to whiten per frequency. We have a different CSD at each frequency, and thus a different beamformer at each frequency. It's just a convenience feature that we pack multiple CSDs into one object and multiple beamformers in a single datastructure.

However, there is now a lot of redundant work happening within `_prepare_beamformer_input`. I plan to refactor that function to unpack the different things it does.

Reply:
Well I do understand we get different weights at each frequency. The question is whether we want to have different lead fields, too. Probably not. And that raises the general question why we whiten lead fields - we do not invert the lead field in beamforming as compared to MNE solutions.

Reply:
I don't know what happens if the lead field is not whitened - if it mathematically would still pan out. Maybe worth a try! AFAIK there is no literature on lead field whitening with beamformers, only covariance matrices - but I might be missing that paper ;)

Reply:
> And that raises the general question why we whiten lead fields - we do not invert the lead field in beamforming as compared to MNE solutions.

At least for MNE any linear operator you apply to the data you should apply to the lead fields because otherwise the results will be biased. I assume the same would hold for beamformers, no?

For example if you apply a projector to the data, you apply it to the lead fields. If you know that this is the correct thing mathematically to do for beamformers when you apply a projection operation to the data, then it is probably mathematically also the correct thing to do in the case of whitening as well. (The whitener implicitly also applies the projection matrix FWIW.)

To me it's not so much about having "different lead fields" as it is about applying the same operators to data and the lead fields before doing computations on them. But maybe I'm thinking about it the wrong way...

Reply:
Well, it is def. the case that whatever is done on the original data needs to be done on the lead field as well. However - does the data need to be whitened to take care of scaling or is it enough to do the covariance matrix? 
I quickly discussed things with @sarangnemo and he shares my intuition about this maybe not being ideal. I will try to run some data through differently whitened beamformers and report back!

Reply:
in the DICS code, there is no data, there is only the CSD matrix.

Reply:
Here is a comparison between whitening both leadfield and the CSD (left) and only whitening the CSD (right):

![fig](https://user-images.githubusercontent.com/428273/68575755-7216c100-0475-11ea-89c0-793cd5b6df97.png)